app/controllers/account/permissions_controller.rb

```diff
@@ -22,6 +22,7 @@ def update
     authorize [:account, @application], :edit_permissions?
 
     current_user.replace_application_permissions(@application, params[:application][:permissions])
+    flash[:success] = @application.id
```

Comment:
If we're going to store an ID in the flash then I think we should probably give it quite a specific key. Otherwise it's feasible that some other future action might populate the `success` with something other than an ID (e.g. with a string), and redirect to `account_applications_path` at which point the template will error in the `message_for_success` message. I half wonder whether there's also a risk that the application has been removed between populating the flash and then calling `message_for_success` in the subsequent request. If you don't want to store the full HTML in the flash then maybe storing the application name and set of permissions might be a bit more robust?